subprojects/gradle/module-dependencies-graph/src/main/kotlin/com/avito/module/configurations/ConfigurationType.kt

```diff
@@ -0,0 +1,54 @@
+package com.avito.module.configurations
+
+import org.gradle.api.artifacts.Configuration
+
+public sealed class ConfigurationType {
+
+    public object Main : ConfigurationType() {
+        override fun toString(): String {
+            return "ConfigurationType.Main"
+        }
+    }
+
+    public object UnitTests : ConfigurationType() {
+        override fun toString(): String {
+            return "ConfigurationType.UnitTests"
+        }
+    }
+
+    public object AndroidTests : ConfigurationType() {
+        override fun toString(): String {
+            return "ConfigurationType.AndroidTests"
+        }
+    }
+
+    public object Lint : ConfigurationType() {
+        override fun toString(): String {
+            return "ConfigurationType.Lint"
+        }
+    }
+
+    public companion object {
+
+        public fun of(configuration: Configuration): ConfigurationType {
+            return when {
+                configuration.isTest() -> UnitTests
+                configuration.isAndroidTest() -> AndroidTests
+                configuration.isLint() -> Lint
+                /**
+                 * TODO Now we define all unknown configurations as Main.
+                 *  But should ignore or convert them to the concrete type
+                 */
+                else -> Main
+            }
+        }
+
+        public fun values(): Set<ConfigurationType> {
+            return setOf(Main, UnitTests, AndroidTests, Lint)
+        }
+
+        private fun Configuration.isTest() = name.startsWith("test")
```

Comment:
The behavior is unchanged

Reply:
What happened? Can't we use hierarchy of configurations?

Reply:
I will create a task